lib/system/comparisons.nim

```diff
@@ -252,11 +250,12 @@ proc max*[T](x: openArray[T]): T =
 
 
 proc clamp*[T](x, a, b: T): T =
-  ## Limits the value ``x`` within the interval [a, b].
-  ##
-  ## .. code-block:: Nim
-  ##   assert((1.4).clamp(0.0, 1.0) == 1.0)
-  ##   assert((0.5).clamp(0.0, 1.0) == 0.5)
+  ## Limits the value `x` within the interval [a, b].
+  ## This proc is equivalent to `max(a, min(b, x))`.
```

Comment:
edge case, but this is incorrect:
```nim
proc main()=
  let a = 10
  let b = 5
  for x in [2, 7, 12]:
    echo (x,)
    echo clamp(x, a, b)
    echo max(a, min(b, x))

main()
```
prints:
```
(2,)
10
10
(7,)
10
10
(12,)
5
10
```

=> should we add `assert a <= b` ?

Reply:
or we could fix this by using `max(min`  instead.

Reply:
potentially more costly?

Reply:
C++ 
https://en.cppreference.com/w/cpp/algorithm/clamp

Reply:
I will open a follow up PR to document this as numpy documentation states.